sql/sys_vars.cc

```diff
@@ -7625,6 +7626,24 @@ static Sys_var_ulonglong Sys_tf_sequence_table_max_upper_bound(
     GLOBAL_VAR(tf_sequence_table_max_upper_bound), CMD_LINE(REQUIRED_ARG),
     VALID_RANGE(1024, ULLONG_MAX), DEFAULT(1048576), BLOCK_SIZE(1));
 
+static Sys_var_charptr Sys_var_blackbox_filename(
+    "blackbox_filename", "Filename for blackbox log on crash",
+    GLOBAL_VAR(blackbox_filename), CMD_LINE(REQUIRED_ARG),
+    IN_FS_CHARSET, DEFAULT("/tmp/blackbox.log"));
+
+std::size_t blackbox_size;
+
+static bool blackbox_size_update(sys_var *, THD *, enum_var_type) {
+  blackbox.resize(blackbox_size);
+  return false;
+}
+
+static Sys_var_ulonglong Sys_var_blackbox_size(
+    "blackbox_size", "Size of the blackbox ringbuffer",
+    GLOBAL_VAR(blackbox_size), CMD_LINE(REQUIRED_ARG),
+    VALID_RANGE(0, ULLONG_MAX), DEFAULT(0), BLOCK_SIZE(1), NO_MUTEX_GUARD,
+    NOT_IN_BINLOG, ON_CHECK(nullptr), ON_UPDATE(blackbox_size_update));
+
```

Comment:
@dutow the option names are not very clear. Can we use something like memory_log_filename or memory_log_filesize
or mem_log_filename, or mem_log_filesize or shared_mem_log_filename?

Reply:
I'm not sure about changing size to filesize, as this is primarily a memory buffer. We only write it out into a file if the server crashes.

As for changing blackbox to memory_log or mem_log, we can do that, I only used blackbox because that's the name mariadb uses.

Reply:
Did we take the implementation from MariaDB?

Reply:
No. This is only in mariadb enterprise, and they used a shared memory buffer.

Reply:
Great! then we are free to choose whatever we want :+1:

Reply:
I've changed the name to buffered_log. With the ringbuffer / discarding behavior gone, I think that's a good description.